### ISSUE TITLE
Tapping in won't work

### ISSUE THREAD

**Issue by BedirT** [at-karan-sharma/autonomous-desk-esphome#2]
Hi,

Thanks for the documentation and the code. But i am confused about your wiring. I have tried setting up exactly as described but i can't seem to get it working properly. 

I have an ESP32-WROOM (I also tried it with an ESP8266 but couldn't handle that as well). The only difference i can see is that i am tapping into the cable instead of using a separate UART cable. But i cannot see why that would cause any issues. Couple things that seems to be not working for me:
1. If i don't wake up the desk with the keypad, i get no reaction from it with ESP.
2. Even when i get some reaction after the wake up it is super jagged and buggy and barely moves at all.
3. I tried to make sure cabling is the same as the pic, and tried a bunch of other ways as well. 

If you could help i would appreciate it!

**Comment by at-karan-sharma** [comment]
Hi @BedirT,

Thanks for giving this project a try! I wondered whether it was feasible to tap the cable between the keypad and motor controller directly, so it's cool that you're testing this approach. Should be possible I feel.

Points 1 and 2 make me wonder if the ESP32 is getting enough (or too much) power. Some ideas:
1. Can you use a multimeter to confirm the voltage between the 5V tap and the GND tap is indeed 5 volts?
2. When you wake up the desk with the keypad, does the voltage remain at 5V?
3. How did you join the tapping wire to the cable? If you soldered them, is the connection solid (i.e. no cold joints)?
4. If you take out the ESP32 and power it only via USB, is it otherwise stable?

**Comment by BedirT** [comment]
Thanks for the quick reply!

1. Yes, I get 5V for this.
2. Yes the voltage is stable at 5V still.
3. I am using wago type connectors.
4. Do you mean using only the keypad? If so yes it works just as it did before when i remove the esp connections. To be clear even when I use with the ESP the keypad works just fine as intended.

**Comment by at-karan-sharma** [comment]
Hmm wago connectors should be pretty stable, assuming the wires are the right gauge. About point 4, I meant to ask if the ESP32 works reliably with esphome. We can rule out any hardware issues if you try a simple config like blinking an LED while the ESP32 is powered over USB (versus powered by the tapped wires).

Do the colors of the tapped wires line up with this table from [Stefichen5](https://github.com/Stefichen5/AutonomousControl)?
| color  | function |
| ------ | -------- |
| red    | sleep(?) |
| orange | GND      |
| green  | TX       |
| yellow | RX       |
| blue   | 5V       |

A pic of the ESP32 board and the tapped wires would be helpful also in understanding your setup.

Worst case, depending on your comfort level, you can try a logic analyzer to confirm the signals being sent across the wire, like Stefichen5.

**Comment by BedirT** [comment]
Yeah the ESP works with no issues. I tested the board couple times.

The colors match exactly. It is also marked in the keypad.

What do you mean by right gauge? I am using some jumper wires, which i think are 28 Gauge. Do you think that could cause enough interference to distrupt the signal? They are pretty short so i thought that couldn't cause the issue? If you think thats okay ill send the image, if not i can try with lower gauge to see if it works and send a pic if not.

**Comment by at-karan-sharma** [comment]
The gauge being an issue is a long shot for sure, but worth ruling out.

Going back to your point 1, is the ESP offline in the ESPHome dashboard until you interact with the keypad? If you're able to see live logs from the ESP, you can check the raw UART data sent across the wire:
1. Replace the snippet from the README with the contents of `config/desky.yml`.
2. Uncomment this line `// UARTDebug::log_hex(direction, bytes, ':');`
3. Save/Install config and view logs. You should see hex data getting regularly printed every second.

If instead no data is printed until you interact with the keypad, I can only conclude it is an electronics issue.

**Comment by deepakkoirala** [comment]
Is this protocol still supported to newer Autonomous Desk? Anyone tried yet?
